src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation of the Function Logger
+"""
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+import yaml
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from deepsparse.loggers.metric_functions import apply_function
+
+
+__all__ = ["FunctionLogger"]
+
+
+class FunctionLogger(BaseLogger):
```

Comment:
Thoughts on renaming to `FilterMapLogger`? Would be nice to have behavior made more explicit (i feel like i'm gonna always forget that it filters things haha)